examples/client/Locomotion/src/pages/RideHistory/RideCard/index.js

```diff
@@ -74,41 +106,60 @@ export const RideListView = ({
 );
 
 const RideView = ({ ride }) => {
+  const [tip, setTip] = useState(null);
   const map = createRef();
+  const getTip = async () => {
+    const priceCalculation = await getPriceCalculation(ride.priceCalculationId);
+    const tipObj = priceCalculation.additionalCharges.find(charge => charge.chargeFor === 'tip');
+    setTip((tipObj || {}).amount);
+  };
+
+  useFocusEffect(() => {
+    if (ride.priceCalculationId) {
+      getTip();
+    }
+  });
   return (
-    <RideViewContainer>
-      <MapRideViewContainer>
-        <Map
-          disableMarkers={ride.state !== RIDE_STATES.COMPLETED}
-          ref={map}
-          ride={ride}
-        />
-      </MapRideViewContainer>
-      <MainRideViewSectionContainer>
-        <RideTitleCard page ride={ride} />
-        <BlankContainer />
-      </MainRideViewSectionContainer>
-      <StopPointsVerticalViewContainer>
-        <StopPointsVerticalView
-          ride={ride}
-        />
-      </StopPointsVerticalViewContainer>
-      <StopPointsVerticalViewContainer>
-        <RidePaymentDetails
-          payment={ride.payment}
-          priceAmount={ride.priceAmount}
-          priceCurrency={ride.priceCurrency}
-        />
-      </StopPointsVerticalViewContainer>
-      <DriverCardContainer>
-        {ride.driver && ride.state === RIDE_STATES.COMPLETED && (
-        <DriverCard
-          activeRide={false}
-          ride={ride}
-        />
-        )}
-      </DriverCardContainer>
-    </RideViewContainer>
+    <>
+      {tip || tip === undefined ? (
```

Comment:
can tip be null? is it ok

Reply:
i only want it to load once its checked the tip, if there isn't it will be undefined so i put either it has a value or undefined. the starting state is null so i dont want it to load while null. any better way to do it?